pkg/controller/status/status.go

```diff
@@ -79,6 +80,16 @@ type StatusManager interface {
 	SetMetaData(meta *metav1.ObjectMeta)
 }
 
+// This regex matches any characters that are not allowed in a Status Reason
+var reasonInvalidCharacters = regexp.MustCompile(`[^A-Za-z0-9_,:]`)
+
+// This regex matches any characters not allows at the start of a Status Reason
```

Comment:
nit:
"allows" -> "allowed" . Happens 2x.
Also, it's best to end the comments with a period.